File: hd/hd.js
console.log(
  (await stdin())
    .trim()
    .split(/\r?\n/)
    .map(x => x.trim())
    .map(format)
    .join("\n")
)

async function stdin() {
  const result = [];
  for await (const chunk of Bun.stdin.stream()) {
    result.push(Buffer.from(chunk).toString("utf8"));
  }
  return result.join("");
}

function format(s) {
  if (s.length === 0) return "";
  if (s.startsWith("# ")) return s;
  return `# ${s}`;
}

File: hd/rust/hd.rs
use std::io::{self, BufRead};

fn main() -> io::Result<()> {
  let stdin = io::stdin();
  let mut handle = stdin.lock();
  
  loop {
    let mut s = String::new();
    let bytes = handle.read_line(&mut s)?;
    if bytes == 0 {
      break;
    }
    let s = s.trim();
    if s.len() != 0 && !s.starts_with("# ") {
      print!("{}", "# ");
    }
    println!("{}", s);
  }

  Ok(())
}
